301-storage-spaces-direct/nested/deploy-s2d-cluster.json

```diff
@@ -0,0 +1,349 @@
+						"value": "[parameters('imageOffer')]"
+					},
+					"imageSKU": {
+						"value": "[parameters('imageSKU')]"
+					},
+					"imageVersion": {
+						"value": "[parameters('imageVersion')]"
+					}
+                }
+            }
+        },
+		{
+			"type": "Microsoft.Compute/virtualMachines/extensions",
+			"name": "[concat(variables('vmNamePrefix'),string(add(copyindex(),1)),'/s2dPrep')]",
+			"apiVersion": "2015-06-15",
+			"location": "[resourceGroup().location]",
+			"dependsOn": [
+				"virtualMachineLoop"
+			],
+			"copy": {
+				"name": "virtualMachineExtensionLoop",
+				"count": "[sub(parameters('vmCount'),1)]"
+			},
+			"properties": {
+				"publisher": "Microsoft.Powershell",
+				"type": "DSC",
+				"typeHandlerVersion": "2.20",
+				"autoUpgradeMinorVersion": true,
+				"settings": {
+					"modulesUrl": "[variables('s2dPrepModulesURL')]",
+					"configurationFunction": "[variables('s2dPrepFunction')]",
+					"properties": {
+						"domainName": "[parameters('domainName')]",
+						"adminCreds": {
+							"userName": "[parameters('adminUserName')]",
+							"password": "PrivateSettingsRef:adminPassword"
+						}
+					}
+				},
+				"protectedSettings": {
+					"items": {
+						"adminPassword": "[parameters('adminPassword')]"
+					}
+				}
+			}
+		},
+		{
+			"type": "Microsoft.Compute/virtualMachines/extensions",
+			"name": "[concat(variables('vmNamePrefix'),'0/s2dConfig')]",
+			"apiVersion": "2015-06-15",
+			"location": "[resourceGroup().location]",
+			"dependsOn": [
+				"virtualMachineLoop",
+				"virtualMachineExtensionLoop",
+				"[resourceId('Microsoft.Storage/storageAccounts', variables('witnessStorageName'))]"				
+			],
+			"properties": {
+				"publisher": "Microsoft.Powershell",
+				"type": "DSC",
+				"typeHandlerVersion": "2.20",
+				"autoUpgradeMinorVersion": true,
+				"settings": {
+					"modulesUrl": "[variables('s2dConfigModulesURL')]",
+					"configurationFunction": "[variables('s2dConfigFunction')]",
+					"properties": {
+						"domainName": "[parameters('domainName')]",
+						"clusterName": "[variables('clusterName')]",
+						"sofsName": "[parameters('sofsName')]",
+						"shareName": "[parameters('shareName')]",
+						"vmNamePrefix": "[variables('vmNamePrefix')]",
+						"vmCount": "[parameters('vmCount')]",
+						"vmDiskSize": "[parameters('vmDiskSize')]",
+						"witnessStorageName": "[variables('witnessStorageName')]",
+						"witnessStorageKey": {
+							"userName": "PLACEHOLDER-DO-NOT-USE",
+							"password": "PrivateSettingsRef:witnessStorageKey"
+						},
+						"adminCreds": {
+							"userName": "[parameters('adminUserName')]",
+							"password": "PrivateSettingsRef:adminPassword"
+						}
+					}
+				},
+				"protectedSettings": {
+					"items": {
+						"adminPassword": "[parameters('adminPassword')]",
+						"witnessStorageKey": "[listKeys(resourceId('Microsoft.Storage/storageAccounts',variables('witnessStorageName')),variables('apiVersionStorage')).key1]"
```

Comment:
if you update the apiVersion on storage the properties in the new schema are:

.keys[0].value